racket/collects/racket/match/parse.rkt

```diff
@@ -52,6 +53,16 @@
           (rest suffix)
           (if (eq? ddk-size #t) 0 ddk-size)))
 
+(define (make-pregexp-matcher r)
+  #`(lambda (e)
+      (let ([r #,r])
+        (regexp-match
+         (cond
+           [(pregexp? r) r]
+           [(string? r) (pregexp r)]
+           [else (raise-argument-error '#,(current-form-name) "(or/c pregexp? string?)" r)])
```

Comment:
Not sure if `raise-argument-error` is the right thing to use here.

Reply:
This seems ok to me.